File: backend/src/main/java/com/djuancito/reposteria/entidad/Producto.java

package com.djuancito.reposteria.entidad;

import java.math.BigDecimal;
import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "Producto")
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productoId") 
    private Integer productoId;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "descripcion")
    private String descripcion;

    @Column(name = "precioBase") 
    private BigDecimal precioBase;

    @Enumerated(EnumType.STRING)
    @Column(name = "categoria")
    private Categoria categoria; 

    @Column(name = "imagenUrl") 
    private String imagenUrl;

    @Column(name = "personalizable")
    private boolean personalizable; 

    @Enumerated(EnumType.STRING)
    @Column(name = "estado")
    private EstadoProducto estado; 
}

File: frontend/src/app/modelos/producto.ts

export type CategoriaProducto = 'torta' | 'postre' | 'bocadito';
export type EstadoProducto = 'activo' | 'inactivo';

export interface Producto {
  productoId: number;
  nombre: string;
  categoria: CategoriaProducto | string; 
  descripcion?: string; 
  precioBase: number;
  imagenUrl: string;
  cloudinaryPublicId?: string;  
  personalizable: boolean;
  personalizacion?: any; 
  estado?: EstadoProducto | string; 
}